src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php

```diff
@@ -386,6 +389,41 @@ function () {
         $this->assertFalse($form->get('field2')->isValid());
         $this->assertCount(1, $form->get('field2')->getErrors());
     }
+
+    public function testDisabledFormIsInvalidIfRootFormValidationFails()
+    {
+        $form = $this->formFactory
+            ->createBuilder(FormType::class, ['field1' => 0], [
+                'constraints' => new Callback(['callback' => static function ($data, ExecutionContextInterface $context) {
+                    if ($data['field1'] < 1) {
+                        $context->addViolation('Invalid');
+                    }
+                }]),
+            ])
+            ->setDisabled(true)
+            ->add('field1')
+            ->getForm();
+
+        $form->submit(null);
+
+        $this->assertTrue($form->isSubmitted());
+        $this->assertFalse($form->isValid());
+        $this->assertCount(1, $form->getErrors());
+    }
+
+    public function testDisabledFormIsInvalidIfChildrenValidationFails()
+    {
+        $form = $this->formFactory->createBuilder(FormType::class, ['field1' => 0])
+            ->setDisabled(true)
+            ->add('field1', null, ['constraints' => new GreaterThanOrEqual(1)])
```

Comment:
It seems this kind of validation (tied directly to the form field) doesn't work because form fields will require to be marked as submitted too:
https://github.com/symfony/symfony/blob/4e0b7e5d5b7ae5c95b8d23e7ab0da2c9ca78d0f2/src/Symfony/Component/Form/Extension/Validator/Constraints/FormValidator.php#L85-L86

and currently only the root form is being marked as "submitted".